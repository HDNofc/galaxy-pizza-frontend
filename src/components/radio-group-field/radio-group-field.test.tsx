import userEvent from '@testing-library/user-event';
import renderWithThemeProvider from 'test-utils/render-with-theme-provider';
import RadioGroupField from './radio-group-field';

const defaultProps = {
  value: 'orange',
  name: 'fruits',
  children: 'радио 1',
  onChange: () => {},
};

describe('<RadioGroupField>', () => {
  it('отображается корректно', () => {
    const { container } = renderWithThemeProvider(<RadioGroupField {...defaultProps} />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('переключается корректно', () => {
    const { getByRole } = renderWithThemeProvider(<RadioGroupField {...defaultProps} />);

    const radioInput = getByRole('radio');
    expect(radioInput).not.toBeChecked();

    userEvent.click(radioInput);
    expect(radioInput).toBeChecked();
  });

  it('коллбек вызывается с корректными параметрами', () => {
    const onChange = jest.fn();
    const { getByRole } = renderWithThemeProvider(
      <RadioGroupField {...defaultProps} onChange={onChange} />
    );
    const radioInput = getByRole('radio');

    userEvent.click(radioInput);
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('проп `disabled` применяется корректно', () => {
    const { getByRole } = renderWithThemeProvider(<RadioGroupField {...defaultProps} disabled />);
    const radioInput = getByRole('radio');

    userEvent.click(radioInput);
    expect(radioInput).toBeDisabled();
    expect(radioInput).not.toBeChecked();
  });

  it('проп `id` применяется корректно', () => {
    const { getByRole, getByLabelText } = renderWithThemeProvider(
      <RadioGroupField {...defaultProps} id="simple-id" />
    );

    expect(getByRole('radio')).toHaveAttribute('id', 'simple-id');
    expect(getByLabelText(/радио 1/i)).toHaveAttribute('id', 'simple-id');
  });
});
