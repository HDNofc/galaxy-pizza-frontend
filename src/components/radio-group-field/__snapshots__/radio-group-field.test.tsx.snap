// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroupField> отображается корректно 1`] = `
<div
  class="RadioGroupField-sc-18y3buo-0 dsLCSU"
>
  <input
    class="Input-sc-18y3buo-2 hKbzCX"
    name="fruits"
    type="radio"
    value="orange"
  />
  <label
    class="Label-sc-18y3buo-1 ceQMEQ"
  >
    радио 1
  </label>
</div>
`;
